blockchain/stake/staketx.go

```diff
@@ -1313,9 +1313,19 @@ func FindSpentTicketsInBlock(block *wire.MsgBlock) *SpentTicketsInBlock {
 	revocations := make([]chainhash.Hash, 0, block.Header.Revocations)
 
 	for _, stx := range block.STransactions {
-		// This can only be called with treasury disabled.
-		const noTreasury = false
-		if IsSSGen(stx, noTreasury) {
+		// It is safe to use the version as a proxy for treasury activation here
+		// since the format of version 3 votes was invalid prior to its
+		// activation and so even if there were votes with version >= 3 prior to
+		// that point, those votes still would've had to conform to the rules
+		// that existed at that time and therefore could not have simultaneously
+		// had the new format and made it into a block.  Further, the old format
+		// is still treated as a valid vote when the flag is set, so the
+		// historical consensus rules would not be violated by incorrectly
+		// setting this flag even if there were votes with version >= 3 prior to
+		// treasury activation.  Finally, there were no votes with versions >= 3
+		// prior to the treasury activation point on mainnet anyway.
+		isTreasuryEnabled := stx.Version >= 3
+		if IsSSGen(stx, isTreasuryEnabled) {
```

Comment:
The issue happened because this returned false for the treasury spend votes?

Reply:
Correct.  Then it made it into the ticket database structures and block index used to service the RPCs.

Reply:
Amazing how much code it took to fix the db after this tiny bug (code wise). Humbling.

Reply:
Reminds me why I prefer positive boolean names in code - to avoid bits like `noTreasury = false` which take extra cycles to understand.